gui/executionGraphGui/client/apps/eg/src/app/components/tools/socket-connection-tool/socket-connection-tool.component.ts

```diff
@@ -42,28 +43,46 @@ export class SocketConnectionToolComponent extends ToolComponent implements OnIn
 
   ngOnInit() {
     this.socketEvents.onDragStart.subscribe(e => {
-      if(e.button!==MouseButton.Left) { return; }
+      if (e.button !== MouseButton.Left) {
+        return;
+      }
       const socket = e.element;
 
       this.logger.info(`Initiating new connection from ${socket.id}`);
       this.sourceSocket = socket;
       if (fromSocket.isOutputSocket(this.sourceSocket)) {
-        this.tempTargetSocket = fromSocket.createSocket('input', socket.typeIndex, Long.fromNumber(0), Long.fromNumber(-1), 'Dummy');
+        this.tempTargetSocket = fromSocket.createSocket(
+          'input',
+          socket.typeIndex,
+          0,
+          Guid.create().toString(),
+          'Dummy'
+        );
       } else {
-        this.tempTargetSocket = fromSocket.createSocket('output', socket.typeIndex, Long.fromNumber(0), Long.fromNumber(-1), 'Dummy');
+        this.tempTargetSocket = fromSocket.createSocket(
+          'output',
+          socket.typeIndex,
+          0,
+          Guid.create().toString(),
```

Comment:
Extract this into a createNodeId() method?

Reply:
Jup, probably better